src/js_realm.hpp

```diff
@@ -1382,41 +1390,200 @@ void RealmClass<T>::compact(ContextType ctx, ObjectType this_object, Arguments&
     return_value.set(realm->compact());
 }
 
+/**
+ * @brief Helper function for `writeCopyTo()` -- parses and validated parameters in a config structure passed from JS
+ *
+ * @param ctx JS context
+ * @param this_object JS's object holding the `RealmClass`
+ * @param args Arguments passed to `writeCopyTo()` from JS
+ * @return realm::Realm::Config A new `Realm::Config` holding the properties of the object passed from JS
+ */
 template <typename T>
-void RealmClass<T>::writeCopyTo(ContextType ctx, ObjectType this_object, Arguments& args, ReturnValue& return_value)
+realm::Realm::Config RealmClass<T>::write_copy_to_helper(ContextType ctx, ObjectType this_object, Arguments& args)
 {
-    args.validate_maximum(2);
+    /* Validation rules:
+     * 1) only one parameter
+     * 2) args[0] must be an object
+     * 3) args[0].path must be present and be a string
+     * 4) args[0].encryptionKey may be present, and must be a binary if it is present
+     * 5) args[0].sync may be present, and must be an object if it is present
+     */
 
-    if (args.count == 0) {
-        throw std::runtime_error("At least path has to be provided for 'writeCopyTo'");
+    // validate 1)
+    if (args.count > 1) {
```

Comment:
What if there is zero arguments? Will `args[0]` fail? Or can we assume that `writeCopyTo()` is validating?

You could use `validate_argument_count(args.count, 1, "`writeCopyTo(<config>)` accepts only one parameter")` to avoid the `if` and `throw`.

Reply:
Done :+1: